core/secrets/createsecret.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2021 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package secrets
+
+import (
+	"encoding/base64"
+	"fmt"
+	"strings"
+
+	"github.com/juju/errors"
+)
+
+// SecretData holds secret key values.
+type SecretData map[string]interface{}
+
+// CreatSecretData creates a secret data bag from a list of arguments.
+// The arguments are either all key=value or a singular value.
+// If base64 is true, then the supplied value(s) are already base64 encoded,
+// otherwise the values are base64 encoded as they are added to the data bag.
+func CreatSecretData(asBase64 bool, args []string) (SecretData, error) {
+	data := make(SecretData)
+	haveSingularalue := false
+	for i, val := range args {
+		// Remove any base64 padding ("=") before splitting the key=value.
+		stripped := strings.TrimRight(val, string(base64.StdPadding))
```

Comment:
I like it because it is easy to use for the end user (IMO)
`create-secret cert=value key=value2`
or
`create-secret apitoken`

'key0 value0 key` value1` is suboptimal because it's hard to distinguish keys vs values

The common case is a single value
I like avoiding having to use `--map` or whatever as it's one extra arg to have to remember
What don't you like about the UX?

Reply:
It's not the UX it's the implementation that I dislike.

Reply:
We need to allow for a single arg that could be "apitoken" or "foo=bar". 

ie
create-secret apitoken
-> {"data": "apitoken"}
create-secret foo=bar 
-> {"foo": "bar"}

That's where the parsing complexity comes in really.
I like the UX like that because it works without requiring --map or --single or whatever.